src/view/draggable_label.py

```diff
@@ -0,0 +1,87 @@
+"""
+A draggable QLabel.
+"""
+
+import enum
+
+from typing import Optional
+
+from PyQt6 import QtWidgets, QtGui, QtCore
+
+
+class DraggableLabel(QtWidgets.QLabel):
+    """
+    A QLabel that is draggable.
+
+    Implements mouseMoveEvent & mousePressEvent.
+    """
+
+    MIME_FORMAT = "application/drag-n-drop-data"
+    """
+    The MIME format to use for Drag and Drop data.
+    """
+
+    class LabelTypes(enum.IntEnum):
+        """
+        Enum of possible labels.
+        """
+
+        POINT_CHARGE = enum.auto()
+        INFINITE_LINE_CHARGE = enum.auto()
+
+        INVALID = enum.auto()
+
+    def __init__(self, parent: Optional[QtWidgets.QWidget]):
+        """
+        Initialize class variables.
+
+        Args:
+            parent (Optional[QWidget]): The parent widget that this widget is a child widget of.
+        """
+
+        super().__init__(parent)
+
+        self.drag_start_pos = QtCore.QPoint()
+
+        self.label_type = DraggableLabel.LabelTypes(DraggableLabel.LabelTypes.INVALID)
+
+    def mousePressEvent(self, ev: QtGui.QMouseEvent) -> None:  # pylint: disable=invalid-name
+        """
+        When the left mouse button is pressed, mark the beginning of the drag.
+
+        Args:
+            ev (QMouseEvent): The QMouseEvent indicating a mouse press.
+        """
+
+        if ev.button() == QtCore.Qt.MouseButton.LeftButton:
+            self.drag_start_pos = ev.pos()
+
+    def mouseMoveEvent(self, ev: QtGui.QMouseEvent) -> None:  # pylint: disable=invalid-name
+        """
+        When the mouse is moved, update its position.
+
+        Args:
+            ev (QMouseEvent): The QMouseEvent indicating a mouse movement.
+        """
+
+        if not ev.buttons() & QtCore.Qt.MouseButton.LeftButton:
+            return
+
+        if ((ev.pos() - self.drag_start_pos).manhattanLength() <
+                QtWidgets.QApplication.startDragDistance()):
+            # Haven't moved far enough yet to initiate a drag
+            return
+
+        data = QtCore.QByteArray()
+
+        val: int = int(self.label_type.value)
```

Comment:
Casted to int because vscode was afraid of calline `to_byte()` on an object of type `any`